### DISCUSSION TITLE
When analyzing SMARTseq2 (iCell8) data using Seurat should one use TPM?

### DISCUSSION THREAD

**Discussion by nikofleischer** [satijalab/seurat#5631] [Q&A]
Hi, I am working extensively on SMARTseq2 single-cell data generated using the Takara iCell8 platform. However, I prefer to work in Seurat instead of their own shiny-based analysis package called CogentDS.

Today I was wondering about which normalization to use. After a long stretch of fruitless googling I couldn't find any useful resource on this.

So far I've been importing raw count data to Seurat. However, SMARTseq2 does not include UMIs. It being full length additionally brings the factor of gene length. From what I've read, most people advise against using TPM data in Seurat as some of its functions are optimized for raw count data. But does this make sense? Am I not introducing heavy sequencing depth (multiple chips with different amounts of cells on each) and gene length (detecting an awful lot of genes with extremely varying length) bias into the analysis? Should I not be working with TPMs instead? Is there a way to generate TPM from counts directly in Seurat if I have gene information available or would I have to do the conversion manually?

Thanks a lot, Niko

**Comment by Carolina-Toste** [comment]
Hello. Did you find a solution to your problem? Icell8 is not used by many people so most tools/ vignettes etc assume that your data has UMIs and were designed for UMI data. If you're comparing genes in the same cell, I do think you should use TPM to correct for gene length. You will need to do that before creating a Seurat object as Seurat does not support gene-length normalization methods. Then you don't run NormalizeData as usual, because you already normalized for gene length /sequencing depth but log transform your TPMs (log1p). If alternatively you want to run SCTransform, you need to do a quasi-UMI transformation (of your TPM counts) first and treat the results as UMI-counts which you can use SCTransform on. On a completely different note as a fellow icell8 user, have you had issues with large batch effects due to chip? Weird residue on chips left from the RT film when chips were removed from the -80?

**Reply by nikofleischer** [reply]
Hi, 
yes I believe I did a manual TPM transformation before loading it into Seurat. I would not treat the counts as UMIs, PCR amplification bias introduced is probably substantial. But I believe more recent Smartseq kits are available with UMIs as well. 

Batch effects due to chips are indeed substantial. Ideally, you would mix multiple samples from different conditions onto the same chip, but I understand that this is difficult to achieve in the lab. I would at least use ample spike in positive controls and don't forget to sequence some empty wells that contain no cells, but have been loaded with the cell suspension to gauge free RNA. 

I did not participate in the wet lab side of things, so I can't say anything about the chips, sorry! 

best of luck with your experiments!
